### PR TITLE
Subs 609 aap merge

### PR DESCRIPTION
**Pull request by neilgoodgame**
Merged in the AAP changes.  By default the tests use a local authentication provider TestWebSecurityConfig.class which creates two test accounts "usi_user" and "usi_admin" . There is also an extension of the APIIntegration tests which uses the AAP as the authentication provider with the test account subs-test.

### CODE REVIEW

**Comment by neilgoodgame** [comment]
I've added a new spring parameter aap.subs-api.secured.  Setting it to false should disable AAP.

File: subs-repository/src/main/java/uk/ac/ebi/subs/repository/security/TeamNameExtractor.java [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+
+    public static final String ADMIN_USER_DOMAIN_NAME = "usiAdmin";
+
+
+    public String adminRole() {
+        return ADMIN_USER_DOMAIN_NAME;
+    }
+
+
+    public String processingStatusTeam(ProcessingStatus processingStatus) {
```
Comment (ke4) [inline subs-repository/src/main/java/uk/ac/ebi/subs/repository/security/TeamNameExtractor.java]: Is this a better method name: teamNameByProcessingStatus?

File: subs-repository/src/main/java/uk/ac/ebi/subs/repository/security/TeamNameExtractor.java [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+                .filter(Objects::nonNull)
+                .findFirst();
+
+        if (!optionalTeam.isPresent()) {
+            throw new IllegalStateException("Cannot find owning team for processing status during authorization " + processingStatus.toString());
+        }
+        return optionalTeam.get().getName();
+    }
+
+    public String submissionStatusTeam(SubmissionStatus submissionStatus) {
```
Comment (ke4) [inline subs-repository/src/main/java/uk/ac/ebi/subs/repository/security/TeamNameExtractor.java]: teamNameBySubmissionStatus ?

File: subs-repository/src/main/java/uk/ac/ebi/subs/repository/security/TeamNameExtractor.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+        Submission submission = submissionRepository.findBySubmissionStatusId(submissionStatus.getId());
+
+        if (submission == null) throw new ResourceNotFoundException();
+
+        return submission.getTeam().getName();
+    }
+
+    public String submissionIdTeam(String submissionId) {
```
Comment (ke4) [inline subs-repository/src/main/java/uk/ac/ebi/subs/repository/security/TeamNameExtractor.java]: teamNameBySubmissionId ?

File: subs-repository/src/test/java/uk/ac/ebi/subs/repository/TeamNameExtractorTest.java [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        submissionStatusRepository.insert(submission.getSubmissionStatus());
+        submissionRepository.insert(submission);
+
+
+        String submissionIdTeam = teamNameExtractor.submissionIdTeam(submission.getId());
+
+        assertThat(submissionIdTeam, is(equalTo(teamName)));
+    }
+
+
```
Comment (ke4) [inline subs-repository/src/test/java/uk/ac/ebi/subs/repository/TeamNameExtractorTest.java]: You could add some negative tests returning `IllegalStateException` and `ResourceNotFoundException `.

File: subs-api/build.gradle [hunk 5]
```diff
@@ -8,7 +8,11 @@ apply plugin: 'application'
 mainClassName = "uk.ac.ebi.subs.ApiApplication"
 
 dependencies {
-    compile project(':subs-repository')
+    compile(project(':subs-repository')) {
+//        exclude module: 'org.springframework.boot:spring-boot-starter-security:1.5.1.RELEASE'
+//        exclude module: 'org.springframework.security:spring-security-config'
```
Comment (neilgoodgame) [inline subs-api/build.gradle]: done

File: subs-api/src/main/java/uk/ac/ebi/subs/api/config/SubsAAPWebSecurityConfig.java [hunk 6]
```diff
@@ -0,0 +1,86 @@
...
+                // don't create session
+                .sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS).and()
+                .authorizeRequests().antMatchers("/api").permitAll()
+                .antMatchers("/").permitAll()
+                .antMatchers("/browser/**/*").permitAll()
+                .antMatchers("/docs/**/*").permitAll()
+                .anyRequest().authenticated();
+
+        //httpSecurity.csrf().disable().sessionManagement().sessionCreationPolicy(SessionCreationPolicy.STATELESS).and().authorizeRequests().anyRequest().permitAll();
+
```
Comment (neilgoodgame) [inline subs-api/src/main/java/uk/ac/ebi/subs/api/config/SubsAAPWebSecurityConfig.java]: done

File: subs-api/src/main/resources/logback.xml [hunk 7]
```diff
@@ -0,0 +1,20 @@
...
+                %d %p [%c] - &lt;%m&gt;%n
+            </pattern>
+        </encoder>
+        <filter class="ch.qos.logback.classic.filter.ThresholdFilter">
+            <level>DEBUG</level>
+        </filter>
+    </appender>
+
+    <logger name="log4j.logger.org.springframework.security" level="DEBUG"/>
+    <logger name="org.springframework.security" level="DEBUG"/>
```
Comment (neilgoodgame) [inline subs-api/src/main/resources/logback.xml]: already done

File: subs-api/src/test/java/uk/ac/ebi/subs/api/TestWebSecurityConfig.java [hunk 8]
```diff
@@ -0,0 +1,78 @@
...
+        return manager;
+    }
+
+    @Override
+    public void configure(WebSecurity web) throws Exception {
+        web.expressionHandler(new DefaultWebSecurityExpressionHandler() {
+            @Override
+            protected SecurityExpressionOperations createSecurityExpressionRoot(Authentication authentication, FilterInvocation fi) {
+                WebSecurityExpressionRoot root = (WebSecurityExpressionRoot) super.createSecurityExpressionRoot(authentication, fi);
+                //root.setDefaultRolePrefix(""); //remove the prefix ROLE_
```
Comment (neilgoodgame) [inline subs-api/src/test/java/uk/ac/ebi/subs/api/TestWebSecurityConfig.java]: Nice to have the debug spring security in the test logs

File: subs-repository/src/main/java/uk/ac/ebi/subs/repository/security/TeamNameExtractor.java (hunk 1 above)
Comment (neilgoodgame) [inline subs-repository/src/main/java/uk/ac/ebi/subs/repository/security/TeamNameExtractor.java]: Dave did this, lets leave it be for now we can always revisit later if there is a problem

File: subs-repository/src/main/java/uk/ac/ebi/subs/repository/security/TeamNameExtractor.java (hunk 2 above)
Comment (neilgoodgame) [inline subs-repository/src/main/java/uk/ac/ebi/subs/repository/security/TeamNameExtractor.java]: Dave did this, lets leave it be for now we can always revisit later if there is a problem

File: subs-repository/src/main/java/uk/ac/ebi/subs/repository/security/TeamNameExtractor.java (hunk 3 above)
Comment (neilgoodgame) [inline subs-repository/src/main/java/uk/ac/ebi/subs/repository/security/TeamNameExtractor.java]: Dave did this, lets leave it be for now we can always revisit later if there is a problem

File: subs-repository/src/test/java/uk/ac/ebi/subs/repository/TeamNameExtractorTest.java (hunk 4 above)
Comment (neilgoodgame) [inline subs-repository/src/test/java/uk/ac/ebi/subs/repository/TeamNameExtractorTest.java]: Dave did this, lets leave it be for now we can always revisit later if there is a problem

**Pull request merged**